scapy/layers/dns.py

```diff
@@ -88,6 +88,30 @@ def dns_get_str(s, p, pkt=None, _internal=False):
     return name, p, s[burned:]
 
 
+def dns_encode(x, check_built=False):
```

Comment:
All callers have check_built set to True, so having this parameter is maybe not needed?

Reply:
I feel that as well as `dns_get_str`, the function should be available as part of the API.

Therefore, I’m turning off by default parameters that wouldn’t make much sense out of scapy’s context (a person calling `dns_encode` doesn’t want his string back when some conditions are met)